mozilla/blurts-server
# add GA4 un/subscribe events

# References:

Jira: MNTOR-3007



# Description

This change adds a custom `unsubscription` and `subscription` event to GA using the [GA4 Measurement Protocol](https://developers.google.com/analytics/devguides/collection/protocol/ga4).

Note that we explored using the standard  [purchase](https://developers.google.com/analytics/devguides/collection/ga4/set-up-ecommerce) event but we don't have all the required parameters we need such as the `value`, since this is fired in response to a webhook and we can't query SubPlat for more information.

# How to test

There is a new `sendPingToGA()` server function which is called in the Accounts webhook handler when subscription notifications are received. This function can also be called directly (see the unit test for an example).

When running locally, `debug_mode` will be set to `true` and pings can be tested as documented at https://developers.google.com/analytics/devguides/collection/protocol/ga4/verify-implementation?client_type=gtag

It's also possible to send pings to the validation server which checks them for well-formedness and returns JSON, but isn't very useful otherwise since you can't inspect the data: https://developers.google.com/analytics/devguides/collection/protocol/ga4/validating-events?client_type=gtag

# Checklist (Definition of Done)
